src/v/cluster/controller_api.cc

```diff
@@ -455,4 +456,79 @@ controller_api::shard_for(const model::ntp& ntp) const {
     return _shard_table.local().shard_for(ntp);
 }
 
+std::vector<model::node_id> replica_set_union(
+  const std::vector<model::broker_shard>& lhs,
+  const std::vector<model::broker_shard>& rhs) {
+    std::vector<model::node_id> unique_replicas;
+    for (const auto& bs : lhs) {
+        auto it = std::find(
+          unique_replicas.begin(), unique_replicas.end(), bs.node_id);
+        if (it == unique_replicas.end()) {
+            unique_replicas.push_back(bs.node_id);
+        }
+    }
+    for (const auto& bs : rhs) {
+        auto it = std::find(
+          unique_replicas.begin(), unique_replicas.end(), bs.node_id);
+        if (it == unique_replicas.end()) {
+            unique_replicas.push_back(bs.node_id);
+        }
+    }
+    return unique_replicas;
+}
+
+ss::future<global_reconciliation_state>
+controller_api::get_global_reconciliation_state(
+  std::vector<model::ntp> ntps, model::timeout_clock::time_point timeout) {
+    // step is to gather per node requests for each of the node
+    absl::node_hash_map<model::node_id, std::vector<model::ntp>> grouped_ntps;
+    const auto ntps_sz = ntps.size();
+
+    for (auto& ntp : ntps) {
+        auto it = _topics.local().updates_in_progress().find(ntp);
+        if (it == _topics.local().updates_in_progress().end()) {
+            // not longer updating
+            continue;
+        }
+        auto all_replicas = replica_set_union(
+          it->second.get_previous_replicas(), it->second.get_target_replicas());
+        for (auto& r : all_replicas) {
+            grouped_ntps[r].push_back(ntp);
+        }
+        co_await ss::maybe_yield();
+    }
+
+    using ret_t = result<std::vector<ntp_reconciliation_state>>;
+
+    auto node_results = co_await ssx::parallel_transform(
```

Comment:
was the maybe_yield used because the set could be large? if that's true, then should concurrently be limited for the parallel transform?